### PR TITLE
Transitioned from Enums to Types. Updated some fields to optional whe…

### PR DESCRIPTION
**Pull request by tcarrio**
…re they may differ situationally (e.g. create/update/request/response). Filled in some TODOs

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by tcarrio** [comment]
Looking for peer review prior to submitting this to upstream.

File: types/dialogflow/index.d.ts [hunk 1]
```diff
@@ -1200,7 +1214,10 @@ export interface QueryParams {
     sentimentAnalysisRequestConfig?: any;
```
Comment (dyladan) [inline types/dialogflow/index.d.ts]: use your interface here

File: types/dialogflow/index.d.ts [hunk 2]
```diff
@@ -1160,7 +1166,11 @@ export interface InputAudioConfig {
...
 
-// TODO export enum OutputAudioEncoding
+export type OutputAudioEncoding =
+    | "OUTPUT_AUDIO_ENCODING_UNSPECIFIED"
+    | "OUTPUT_AUDIO_ENCODING_LINEAR_16"
+    | "OUTPUT_AUDIO_ENCODING_MP3"
+    | "OUTPUT_AUDIO_ENCODING_OGG_OPUS"
 
 export interface OutputAudioConfig {
     audioEncoding: any;
```
Comment (dyladan) [inline types/dialogflow/index.d.ts]: Use OutputAudioEncoding

File: types/dialogflow/index.d.ts [hunk 3]
```diff
@@ -701,16 +701,16 @@ export interface Status {
     details: any[];
```
Comment (dyladan) [inline types/dialogflow/index.d.ts]: Should there be a type here?

File: types/dialogflow/index.d.ts [hunk 4]
```diff
@@ -905,27 +906,25 @@ export interface ClientOptions {
 }
 
 export interface EntitySynonyms {
-    synonyms: string[];
+    synonyms: string[] & {0: string};
```
[the remark this answers is not in the document]
Comment (dyladan) [reply types/dialogflow/index.d.ts]: `[string] & string[]`

[the remark this answers is not in the document]
Comment (dyladan) [reply types/dialogflow/index.d.ts]:
```
type NonEmptyArray<T> = [T] & T[];

synonyms: NonEmptyArray<string>;
```

**Comment by dyladan** [comment]
@tcarrio make sure you run the test and lint steps before making the PR to definitely typed

File: types/dialogflow/index.d.ts (hunk 3 above)
Comment (tcarrio) [reply types/dialogflow/index.d.ts]: Looks like it should be following this: 

https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Any

File: types/dialogflow/index.d.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (tcarrio) [reply types/dialogflow/index.d.ts]: Updated for review

File: types/dialogflow/index.d.ts (hunk 1 above)
Comment (tcarrio) [reply types/dialogflow/index.d.ts]: Updated for review

File: types/dialogflow/index.d.ts (hunk 2 above)
Comment (tcarrio) [reply types/dialogflow/index.d.ts]: Updated for review

**Pull request closed without merging**